### PR TITLE
snowflake-destination: allow buffer to be increased till 50

### PR DESCRIPTION
**Pull request by subodh1810**
See this slack thread https://airbytehq-team.slack.com/archives/C046KAVD4NR/p1680158784059009

### CODE REVIEW

**Review by edgao — APPROVED** [review]
we should probably prioritize decoupling buffer flushes from checkpoints at some point, but this is a good workaround for now

File: airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeDestinationResolver.java [hunk 1]
```diff
@@ -37,6 +38,16 @@ public static boolean isAzureBlobCopy(final JsonNode config) {
         && config.get("loading_method").has("azure_blob_storage_account_name");
   }
 
+  public static int getNumberOfFileBuffers(final JsonNode config) {
+    int numOfFileBuffers = FileBuffer.DEFAULT_MAX_CONCURRENT_STREAM_IN_BUFFER;
+    if (config.has(FileBuffer.FILE_BUFFER_COUNT_KEY)) {
+      numOfFileBuffers = Math.min(config.get(FileBuffer.FILE_BUFFER_COUNT_KEY).asInt(), FileBuffer.MAX_CONCURRENT_STREAM_IN_BUFFER);
+    }
```
Comment (ryankfu) [inline airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeDestinationResolver.java:45]: Non-blocking but maybe a ticket should be filed to add this warning as well here

https://github.com/airbytehq/airbyte/blob/0b156a8326f5566c14063e424951f58c9495187a/airbyte-integrations/connectors/destination-redshift/src/main/java/io/airbyte/integrations/destination/redshift/RedshiftStagingS3Destination.java#L146-L152

It's to safeguard users from running into OOM issues by logging the the increased memory size that comes with more FileBuffers

**Review by ryankfu — APPROVED** [review]
LGTM, non-blocking comment about a `LOGGER.warn` and maybe add a test but this can be separate

**Comment by dimoschi** [comment]
How can we use that feature to increase the buffer size?

**Comment by subodh1810** [comment]
@dimoschi upgrade snowflake destination to the latest version. Once done go to the snowflake connector setup page and in the UI you will see this option, you can increase it till 50.
Screenshot 2023-03-31 at 12 49 58 AM

**Comment by dimoschi** [comment]
> @dimoschi upgrade snowflake destination to the latest version. Once done go to the snowflake connector setup page and in the UI you will see this option, you can increase it till 50.
> 
> Screenshot 2023-03-31 at 12 49 58 AM
> 
> 

Thanks

**Pull request merged**